watttime/tcy.py

```diff
@@ -0,0 +1,125 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import pandas as pd
+import pytz
+from typing import Optional, List
+from .api import WattTimeHistorical
+import os
+
+@dataclass
+class TCYConfig:
+    """Configuration for TCY calculation"""
+    region: str
+    timezone: str
+    holidays: Optional[List[str]] = None
+
+class TCYCalculator:
+    """Calculate Typical Carbon Year profiles from historical MOER data using a 3-year lookback period"""
+    
+    def __init__(
+        self,
+        region: str,
+        timezone: str,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        holidays: Optional[List[str]] = None
+    ):
+        # Handle credentials like other API classes
+        self.username = username or os.getenv("WATTTIME_USER")
+        self.password = password or os.getenv("WATTTIME_PASSWORD")
+        
+        # Initialize API client with credentials
+        self.wt_client = WattTimeHistorical(self.username, self.password)
```

Comment:
```suggestion
        self.wt_client = WattTimeHistorical(username, password)
```

You can delete the previous lines and just pass the arguments  straight to `WattTimeHistorical`. If they default to `None`, then we pass `None` to `WattTimeHistorical`, which then looks at the getenv. It's not strictly necessary, but it keeps us from repeating the same functionality twice in the code base, so future updates are easier to manage if it's only handled in one location.